### DISCUSSION TITLE
importing data from nextcloud Passwords

### DISCUSSION THREAD

**Discussion by digrouz** [dani-garcia/vaultwarden#2464] [Q&A] [answered]
Hi,

New to vaultwarden here. What would be the best way to import all my passwords from Nextcloud Passwords into vaultwarden ?

I understand I have to play with CSV exports but up to now, i'm unsuccessfull importing such csv into vaultwarden.

Kind regards

Nicolas

**Comment by Masgalor** [comment, marked as answer]
Imports are handled by the client, did you have a look at the official documentation?

https://bitwarden.com/help/import-faqs/#q-what-file-formats-does-bitwarden-support-for-import
https://bitwarden.com/help/condition-bitwarden-import/

**Reply by digrouz** [reply]
I'm trying, but I always get data are not properly formated.

Is there any log somewhere we I could understand what's wrong with my format is the following using the export feature of Nextcloud Passwords:
![export format](https://user-images.githubusercontent.com/548498/169709466-2eca2896-8ed1-4bad-9928-b83a0b764254.png)

Then with spreadsheet, i'm filling 3rd column with "login" so most of my lines look like this:
`"somefolder","0",login,"Label","Notes","","","https://someurl.net","username","password",""`

I even get that error while using sample line from manual
`,,login,Login Name,,,,,,`

But it seems importing process does not like it :(

Any help apreciated.

**Reply by digrouz** [reply]
Is there any log I can check to see what is not properly formated ?

**Reply by BlackDex** [reply]
This is all client side. So nothing on the server side repainted logs. Maybe something in the developer tools of the browser (F12)?

**Reply by digrouz** [reply]
I see some websocket errors, is this required to import data ?

**Reply by BlackDex** [reply]
Nope

**Reply by digrouz** [reply]
might it be because of firefox and/or macos ?

**Reply by BlackDex** [reply]
I have no clue.as mentioned before, this is all done on the client side. Vaultwarden is only the server side.

**Reply by digrouz** [reply]
I did fix the websocket issue I had, now debug console is only showing a warning `"SweetAlert2: "onOpen" is deprecated and will be removed in the next major release. Please use "didOpen" instead."`

**Reply by digrouz** [reply]
ok, got it working by generating a keepass like csv.

**Comment by monperrus** [comment]
Wrote a python script: https://gist.github.com/monperrus/c671817d53f6fc4bfe8d1773f28262d7

**Reply by godisopensource** [reply]
Hey! 

I was interested in migrating my Nextcloud passwords to Bitwarden, and tested your script. It needed some updates so I applied them and posted the updated version on my Github with a README. Feel free to check/improve it!

[https://github.com/godisopensource/np2bw/e](https://github.com/godisopensource/np2bw)

Have a good day, and thank you for having provided a base code!

**Reply by bethnard** [reply]
Hey, I just would like to emphasize that the code from @godisopensource works flawlessly. 
Just one highlight: the code is programmed for French language. So either one sets NC language to French prior to exporting the .csv file from NC Passwords or the programmed code shall be edited and the key words need to be translated to the appropriate language. Thank you!
